tremor-script/src/interpreter/imut_expr.rs

```diff
@@ -60,12 +62,37 @@ where
         meta: &'run Value<'event>,
         local: &'run LocalStack<'event>,
     ) -> Result<Cow<'event, str>> {
-        match stry!(self.run(opts, env, event, state, meta, local)).borrow() {
-            Value::String(s) => Ok(s.clone()),
-            other => error_need_obj(self, self, other.value_type(), &env.meta),
+        match stry!(self.run(opts, env, event, state, meta, local)).into_owned() {
+            Value::String(s) => Ok(s),
+            other => error_need_str(self, self, other.value_type(), &env.meta),
         }
     }
 
+    /// Evaluates the expression to an index, i.e., a `usize`, or returns the appropriate error
+    /// indicating why the resulting `Value` — if any — is not an index.
+    ///
+    /// # Note
+    /// This method explicitly *does not* check whether the resulting index is in range of the array.
+    #[inline]
+    pub fn eval_to_index<Expr>(
+        &'script self,
+        outer: &'run Expr,
+        opts: ExecOpts,
+        env: &'run Env<'run, 'event, 'script>,
+        event: &'run Value<'event>,
+        state: &'run Value<'static>,
+        meta: &'run Value<'event>,
+        local: &'run LocalStack<'event>,
+        path: &'script Path,
+        array: &[Value],
+    ) -> Result<usize>
+    where
+        Expr: BaseExpr,
+    {
+        let val = self.run(opts, env, event, state, meta, local)?;
```

Comment:
this should use  `stry!` instead of `?`

Reply:
there are a few more places about this I'll call out for simplicty.
```suggestion
        let val = stry!(self.run(opts, env, event, state, meta, local));
```